### PR TITLE
[release/3.0] Update gRPC package version to 2.23.2

### PR DESCRIPTION
**Pull request by JamesNK**
Fixes https://github.com/aspnet/AspNetCore/issues/14348

**Description**

Update Grpc.AspNetCore reference in gRPC template to 2.23.2.

**Customer impact**

Template currently references 2.23.1. This package is impacted by the NuGet deterministic issue, which could result in some gRPC runtime files not being deployed.

**Regression**

No

**Risk**

Low

### CODE REVIEW

**Review by dougbu — APPROVED** [review]
WFM if QB approved

**Comment by Pilchie** [comment]
@JamesNK This is approved for 3.0.1. Go ahead and merge when tests are ok.

**Comment by dougbu** [comment]
@JamesNK please do not merge until #14404 is in

**Review by JunTaoLuo — APPROVED** [review]

**Comment by JamesNK** [comment]
Merge button is disable. I'm guessing I don't have permission.

Merge pls 🙏 

Also, will this make it into 3.1?

**Comment by Pilchie** [comment]
I broke it :(  I was going to retarget the base to `release/3.1-preview1` so that it could be in Preview 1, but it picked up more commits, that stayed even when I retargeted back to `release/3.1`.  

Can someone fix the branch and re-target to `release/3.1-preview1` so we can get it in today?

**Comment by dougbu** [comment]
@Pilchie done

All other "reviewers" sorry for the spam. As usual GitHub went rogue between pushing the rebased branch and resetting the PR's target.

**Comment by Pilchie** [comment]
Thanks @dougbu!  Let's merge this once green.

**Review by Pilchie — APPROVED** [review]

**Comment by Pilchie** [comment]
Wait - we put this in `3.1-preview1`, but we wanted to put it in `3.0.x` :(

**Comment by dougbu** [comment]
My bad. Do you want me to push a `revert` commit directly into 'release/3.1-preview1` and then rebase @JamesNK's branch (which I still have locally)?

**Comment by dougbu** [comment]
Or, do we want it in both branches?

**Comment by JamesNK** [comment]
Both branches.

**Pull request merged**
